### DISCUSSION TITLE
PlayCDN default wrapper for custom plugins

### DISCUSSION THREAD

**Discussion by Triloworld** [tailwindlabs/tailwindcss#7668] [Ideas]
Right now injecting `plugin` to `tailwind.config` is hacky and not standard way - special css only on develop site can be created as plugin that enrich DX.

If `plugins` array in imported `tailwind.config.js` get string, with `function()` or `(){}` then threat it correct and import. Maybe external http add or maybe can be somehow required inline from node_modules

Thx @sntran for solution and description current way:

The `tailwind` global object has a `plugin` function that behaves the same as `tailwindcss/plugin`.

I have been using the following tailwind.config.js **file** that can be used with both the Play CDN and the CLI when needed:

```js
let tailwind;
try {
  tailwind = require("tailwindcss");
} catch (_) {
  tailwind = window.tailwind || {};
}

const plugin = tailwind.plugin;

tailwind.config = {
  theme: {
    ...
  },
  plugins: [
    plugin(function siblingVariants({ addVariant, e }) {
      // Adjacent sibling combinator
      addVariant("sibling", "& + *");
      // General sibling combinator
      addVariant("siblings", "& ~ *");
    }),
  ],
};

if (typeof module !== 'undefined' && module.exports) {
  module.exports = tailwind.config;
}
```

And on the HTML:

```html
<script src="https://cdn.tailwindcss.com?plugins=aspect-ratio,line-clamp"></script>
<script src="/css/tailwind.config.js"></script>
```

_Originally posted by @sntran in https://github.com/tailwindlabs/tailwindcss/discussions/6473#discussioncomment-2007720_

---

What I see as solution:

```js
module.exports = {
  theme: {
    ...
  },
  plugins: [
    function ({ addVariant, e }) { // Recognize that this is custom plugin, argument string is https externall one, argument function is inlined like this one
      // Adjacent sibling combinator
      addVariant("sibling", "& + *");
      // General sibling combinator
      addVariant("siblings", "& ~ *");
    }),
  ],
};
```

**Comment by adamwathan** [comment]
Hey thanks for the feedback! Converting this over to a feature request as we don't consider it a bug but totally open to discussing any ideas here.
